src/liboslexec/opmessage.cpp

```diff
@@ -27,33 +27,34 @@ namespace pvt {
 
 
 OSL_SHADEOP void
-osl_setmessage(ShaderGlobals* sg, const char* name_, long long type_, void* val,
-               int layeridx, const char* sourcefile_, int sourceline)
+osl_setmessage(ShaderGlobals* sg, ustring_pod name_, long long type_, void* val,
+               int layeridx, ustring_pod sourcefile_, int sourceline)
 {
-    const ustring& name(USTR(name_));
-    const ustring& sourcefile(USTR(sourcefile_));
+    ustringhash name       = ustringhash_from(USTR(name_));
```

Comment:
Does USTR(...) casting need to be involved anymore?  I thought a benefit of these changes is to not need that very pointer/reinterpret_cast'ish and just use the ustring_rep or ustring_pod?

Reply:
I think in most cases, that's where we'll eventually get to, yes.